components/brave_wallet/common/eth_request_helper.cc

```diff
@@ -365,4 +380,106 @@ bool ParseSwitchEthereumChainParams(const std::string& json,
+  if (!token || !error_message) {
+    return false;
+  }
+  *error_message = "";
+
+  // Might be a list from legacy send method.
+  absl::optional<base::Value> params = GetObjectFromParamsList(json);
+  if (!params)
+    params = GetParamsDict(json);
+
+  if (!params || !params.value().is_dict()) {
+    *error_message = "params parameter is required";
+    return false;
+  }
+
+  const std::string* type = params.value().FindStringKey("type");
+  if (!type) {
+    *error_message = "type parameter is required";
+    return false;
+  }
+  // Only ERC20 is supported currently.
+  if (*type != "ERC20") {
+    *error_message =
+        base::StringPrintf("Asset of type '%s' not supported", type->c_str());
+    return false;
+  }
+
+  const base::Value* options_dict = params.value().FindDictKey("options");
+  if (!options_dict) {
+    *error_message = "options parameter is required";
+    return false;
+  }
+
+  const std::string* address = options_dict->FindStringKey("address");
+  if (!address) {
+    *error_message = "address parameter is required";
+    return false;
+  }
+
+  const auto eth_addr = EthAddress::FromHex(*address);
+  if (eth_addr.IsEmpty()) {
+    *error_message =
+        base::StringPrintf("Invalid address '%s'", address->c_str());
+    return false;
+  }
+
+  const std::string* symbol = options_dict->FindStringKey("symbol");
+  if (!symbol) {
+    *error_message = "symbol parameter is required";
+    return false;
+  }
+
+  // EIP-747 limits the symbol length to 5, but metamask uses 11, so we use
+  // the same limit here for compatibility.
+  if (symbol->size() == 0 || symbol->size() > 11) {
+    *error_message = base::StringPrintf(
+        "Invalid symbol '%s': symbol length should be greater than 0 and less "
+        "than 12",
+        symbol->c_str());
+    return false;
+  }
+
+  // Allow decimals in both number and string for compability.
+  // EIP747 specifies the type of decimals number, but websites like coingecko
+  // uses string.
+  const base::Value* decimals_value = options_dict->FindKey("decimals");
+  if (!decimals_value ||
+      (!decimals_value->is_int() && !decimals_value->is_string())) {
+    *error_message = "decimals parameter is required.";
+    return false;
+  }
+  int decimals = decimals_value->is_int() ? decimals_value->GetInt() : 0;
+  if (decimals_value->is_string() &&
+      !base::StringToInt(decimals_value->GetString(), &decimals)) {
+    *error_message = base::StringPrintf(
+        "Invalid decimals '%s': decimals should be a number greater than 0 and "
+        "less than 36",
+        decimals_value->GetString().c_str());
+    return false;
+  }
+
+  if (decimals < 0 || decimals > 36) {
+    *error_message = base::StringPrintf(
+        "Invalid decimals '%d': decimals should be greater than 0 and less "
+        "than 36",
+        decimals);
+    return false;
+  }
+
+  // TODO(jocelyn): Parse image URL to be the logo URL once we supports
+  // downloading an icon from a remote URL.
```

Comment:
Let's parse it pls just so it's stored at least for the future when we want to use it. We can put it in `logo` which already exists.

Reply:
https://github.com/brave/brave-core/pull/11518/commits/e8d4fe2f5e0ad9aa2989212b10d59932eea368b8